wa-system/view/waAppViewHelper.class.php

```diff
@@ -61,7 +62,16 @@ public function pages($parent_id = 0, $with_params = true)
             } else {
                 $routes = wa()->getRouting()->getByApp($this->wa->getConfig()->getApplication(), $domain);
                 if ($routes) {
-                    $route = end($routes);
+                    if (is_string($route_url)) {
+                        foreach ($routes as $route) {
+                            if ($route['url'] === $route_url) {
+                                break;
+                            }
+                        }
+                        return array();
+                    } else {
+                        $route = current($routes);
```

Comment:
This must be `end($routes)` to preserve current behaviour: use the last (usualy the most general) route of the app.